src/NodeApi/Interop/JSSynchronizationContext.cs

```diff
@@ -76,8 +108,44 @@ public void OpenAsyncScope()
     /// Decrement reference count for the main loop async resource.
     /// Non-zero count prevents Node.JS process from exiting.
     /// </summary>
-    public void CloseAsyncScope()
+    public override void CloseAsyncScope()
     {
         _tsfn.Unref();
     }
 }
+
+public sealed class JSDispatcherSynchronizationContext : JSSynchronizationContext
+{
+    private readonly JSDispatcherQueue _queue;
+
+    public JSDispatcherSynchronizationContext(JSDispatcherQueue queue)
+    {
+        _queue = queue;
+    }
+
+    public override void Post(SendOrPostCallback callback, object? state)
+    {
+        if (IsDisposed) return;
+
+        _queue.TryEnqueue(() => callback(state));
+    }
+
+    public override void Send(SendOrPostCallback callback, object? state)
+    {
+        if (this == Current)
+        {
+            callback(state);
+            return;
+        }
+
+        if (IsDisposed) return;
+
+        using ManualResetEvent syncEvent = new(false);
+        _queue.TryEnqueue(() =>
+        {
+            callback(state);
+            syncEvent.Set();
+        });
+        syncEvent.WaitOne();
```

Comment:
If `TryEnqueue()` returned `false`, then it shouldn't wait.

Reply:
Good catch! Fixed.